In [2]:
# ============================================================
# 1. IMPORT LIBRARIES & LOAD DATASET
# ============================================================

import pandas as pd
import numpy as np

# Load dataset
df = pd.read_csv("Day8_Ecommerce_Sales_Dataset.csv")

# Display dataset
df.head()

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI


In [3]:
# ============================================================
# 2. DATASET EXPLORATION & BASIC STATISTICS
# ============================================================

print("Dataset Shape:", df.shape)

print("\nColumn Names:")
print(df.columns.tolist())

print("\nIndex:")
print(df.index)

print("\nData Types:")
print(df.dtypes)

print("\nDataset Information:")
df.info()

print("\nFirst 5 Rows:")
display(df.head())

print("\nLast 5 Rows:")
display(df.tail())

print("\nRandom Sample:")
display(df.sample(5))

print("\nMissing Values:")
print(df.isnull().sum())

print("\nDuplicate Rows:")
print(df.duplicated().sum())

print("\nDescriptive Statistics:")
display(df.describe())

Dataset Shape: (100, 13)

Column Names:
['Order_ID', 'Order_Date', 'Customer', 'City', 'Region', 'Category', 'Product', 'Quantity', 'Unit_Price', 'Discount_Percent', 'Total_Sales', 'Rating', 'Payment_Method']

Index:
RangeIndex(start=0, stop=100, step=1)

Data Types:
Order_ID              int64
Order_Date           object
Customer             object
City                 object
Region               object
Category             object
Product              object
Quantity              int64
Unit_Price            int64
Discount_Percent      int64
Total_Sales         float64
Rating              float64
Payment_Method       object
dtype: object

Dataset Information:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Order_ID          100 non-null    int64  
 1   Order_Date        100 non-null    object 
 2   Customer          100 non-null    o

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI



Last 5 Rows:


,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
95,1096,2026-01-26,Aditya Verma,Kochi,South,Sports,Dumbbell Set,5,1999,5,9495.25,4.2,Credit Card
96,1097,2026-02-02,Arjun Nair,Mumbai,West,Clothing,T-Shirt,1,799,15,679.15,3.5,UPI
97,1098,2026-01-04,Yusuf Dar,Bengaluru,South,Home & Kitchen,Electric Kettle,4,1499,0,5996.00,4.2,Net Banking
98,1099,2026-03-06,Hiba Sheikh,Ahmedabad,West,Books,Machine Learning Basics,2,999,0,1998.00,3.8,UPI
99,1100,2026-02-08,Kabir Ali,Pune,West,Books,Data Science Handbook,5,899,0,4495.00,4.8,Debit Card



Random Sample:


,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
53,1054,2026-04-22,Aisha Mir,Chennai,South,Home & Kitchen,Coffee Maker,1,3499,15,2974.15,4.3,Net Banking
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
97,1098,2026-01-04,Yusuf Dar,Bengaluru,South,Home & Kitchen,Electric Kettle,4,1499,0,5996.00,4.2,Net Banking
71,1072,2026-05-03,Aarav Sharma,Jaipur,North,Sports,Yoga Mat,5,899,10,4045.50,4.3,Debit Card
31,1032,2026-01-16,Ishita Gupta,Pune,West,Books,Atomic Habits,2,599,10,1078.20,4.5,UPI



Missing Values:
Order_ID            0
Order_Date          0
Customer            0
City                0
Region              0
Category            0
Product             0
Quantity            0
Unit_Price          0
Discount_Percent    0
Total_Sales         0
Rating              0
Payment_Method      0
dtype: int64

Duplicate Rows:
0

Descriptive Statistics:


,Order_ID,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating
count,100.000000,100.000000,100.000000,100.000000,100.000000,100.000000
mean,1050.500000,2.940000,1647.000000,6.500000,4472.629500,4.322000
std,29.011492,1.391315,970.054669,6.092718,3324.306691,0.414285
min,1001.000000,1.000000,499.000000,0.000000,449.100000,3.500000
25%,1025.750000,2.000000,799.000000,0.000000,1948.050000,4.100000
50%,1050.500000,3.000000,1499.000000,5.000000,3228.300000,4.300000
75%,1075.250000,4.000000,2499.000000,11.250000,6427.562500,4.700000
max,1100.000000,5.000000,3499.000000,15.000000,13995.000000,5.000000


In [4]:
# ============================================================
# 3. DATA SELECTION, FILTERING & SORTING
# ============================================================

# Select specific columns
display(df[["Order_ID", "City", "Category", "Product", "Total_Sales"]].head())

# Select rows using iloc
display(df.iloc[0:10])

# Orders with sales greater than ₹5,000
high_value_orders = df[df["Total_Sales"] > 5000]
print("Orders with Sales > ₹5,000:")
display(
    high_value_orders[
        ["Order_ID", "City", "Product", "Total_Sales"]
    ].sort_values("Total_Sales", ascending=False)
)

# Orders from Mumbai
print("Mumbai Orders:")
display(
    df[df["City"] == "Mumbai"][
        ["Order_ID", "Product", "Category", "Total_Sales"]
    ]
)

# Sports category orders
print("Sports Category Orders:")
display(
    df[df["Category"] == "Sports"][
        ["Order_ID", "Product", "Quantity", "Total_Sales"]
    ]
)

# UPI payment orders
print("UPI Orders:")
display(
    df[df["Payment_Method"] == "UPI"][
        ["Order_ID", "Product", "Total_Sales"]
    ]
)

# Multiple-condition filtering
print("Sports Orders with Sales > ₹5,000:")
display(
    df[
        (df["Category"] == "Sports") &
        (df["Total_Sales"] > 5000)
    ][
        ["Order_ID", "City", "Product", "Total_Sales"]
    ]
)

# Top 10 orders by sales
print("Top 10 Orders by Sales:")
display(
    df.sort_values(
        by="Total_Sales",
        ascending=False
    ).head(10)
)

# Top 10 orders by quantity
print("Top 10 Orders by Quantity:")
display(
    df.sort_values(
        by="Quantity",
        ascending=False
    ).head(10)
)

,Order_ID,City,Category,Product,Total_Sales
0,1001,Hyderabad,Electronics,Wireless Headphones,4497.00
1,1002,Jammu,Sports,Running Shoes,2799.00
2,1003,Chennai,Electronics,Power Bank,2038.30
3,1004,Lucknow,Electronics,Mechanical Keyboard,9496.20
4,1005,Jammu,Electronics,Smart Watch,3134.05


,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.00,3.8,Cash on Delivery
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.00,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.30,4.6,Cash on Delivery
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
4,1005,2026-03-30,Priya Menon,Jammu,North,Electronics,Smart Watch,1,3299,5,3134.05,4.2,UPI
5,1006,2026-01-21,Manya Rao,Kochi,South,Books,The Alchemist,1,499,10,449.10,4.2,Cash on Delivery
6,1007,2026-01-12,Alina Shah,Jaipur,North,Home & Kitchen,Water Bottle,2,699,0,1398.00,4.1,Debit Card
7,1008,2026-04-27,Aman Kumar,Jammu,North,Clothing,Hoodie,4,1599,5,6076.20,4.3,Credit Card
8,1009,2026-06-05,Aman Kumar,Mumbai,West,Home & Kitchen,Electric Kettle,3,1499,0,4497.00,4.0,Cash on Delivery
9,1010,2026-03-11,Aman Kumar,Kochi,South,Clothing,Jacket,4,2499,10,8996.40,4.7,Credit Card


Orders with Sales > ₹5,000:


,Order_ID,City,Product,Total_Sales
48,1049,Mumbai,Running Shoes,13995.00
52,1053,Hyderabad,Running Shoes,13295.25
84,1085,Hyderabad,Coffee Maker,12596.40
89,1090,Lucknow,Cricket Bat,12495.00
12,1013,Lucknow,Jacket,10620.75
47,1048,Lucknow,Coffee Maker,10497.00
79,1080,Delhi,Jacket,9996.00
69,1070,Delhi,Jacket,9996.00
51,1052,Chandigarh,Mechanical Keyboard,9496.20
3,1004,Lucknow,Mechanical Keyboard,9496.20


Mumbai Orders:


,Order_ID,Product,Category,Total_Sales
8,1009,Electric Kettle,Home & Kitchen,4497.00
48,1049,Running Shoes,Sports,13995.00
61,1062,Mechanical Keyboard,Electronics,7497.00
66,1067,The Alchemist,Books,2245.50
78,1079,Hoodie,Clothing,1599.00
93,1094,Mixer Grinder,Home & Kitchen,8547.15
94,1095,Power Bank,Electronics,2398.00
96,1097,T-Shirt,Clothing,679.15


Sports Category Orders:


,Order_ID,Product,Quantity,Total_Sales
1,1002,Running Shoes,1,2799.00
11,1012,Cricket Bat,2,4498.20
13,1014,Running Shoes,3,8397.00
16,1017,Cricket Bat,3,7497.00
20,1021,Yoga Mat,1,809.10
22,1023,Football,3,2157.30
23,1024,Running Shoes,1,2799.00
24,1025,Football,5,3995.00
26,1027,Football,5,3395.75
30,1031,Yoga Mat,1,764.15


UPI Orders:


,Order_ID,Product,Total_Sales
4,1005,Smart Watch,3134.05
10,1011,Coffee Maker,6998.00
13,1014,Running Shoes,8397.00
14,1015,Wireless Headphones,2998.00
15,1016,The Alchemist,1696.60
18,1019,Jacket,7497.00
19,1020,Mixer Grinder,2999.00
23,1024,Running Shoes,2799.00
24,1025,Football,3995.00
27,1028,Wireless Headphones,5696.20


Sports Orders with Sales > ₹5,000:


,Order_ID,City,Product,Total_Sales
13,1014,Jaipur,Running Shoes,8397.00
16,1017,Hyderabad,Cricket Bat,7497.00
37,1038,Pune,Dumbbell Set,9495.25
48,1049,Mumbai,Running Shoes,13995.00
52,1053,Hyderabad,Running Shoes,13295.25
58,1059,Chennai,Cricket Bat,8496.60
59,1060,Pune,Dumbbell Set,5397.30
89,1090,Lucknow,Cricket Bat,12495.00
92,1093,Hyderabad,Cricket Bat,8496.60
95,1096,Kochi,Dumbbell Set,9495.25


Top 10 Orders by Sales:


,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
48,1049,2026-03-02,Ishita Gupta,Mumbai,West,Sports,Running Shoes,5,2799,0,13995.00,4.0,Net Banking
52,1053,2026-01-08,Ananya Singh,Hyderabad,South,Sports,Running Shoes,5,2799,5,13295.25,4.2,Credit Card
84,1085,2026-01-28,Sana Malik,Hyderabad,South,Home & Kitchen,Coffee Maker,4,3499,10,12596.40,4.3,Net Banking
89,1090,2026-05-13,Aditya Verma,Lucknow,North,Sports,Cricket Bat,5,2499,0,12495.00,3.8,Credit Card
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking
47,1048,2026-06-24,Sara Ahmed,Lucknow,North,Home & Kitchen,Coffee Maker,3,3499,0,10497.00,5.0,UPI
69,1070,2026-06-30,Aditya Verma,Delhi,North,Clothing,Jacket,4,2499,0,9996.00,3.8,Net Banking
79,1080,2026-05-25,Aditya Verma,Delhi,North,Clothing,Jacket,4,2499,0,9996.00,4.6,Debit Card
51,1052,2026-03-13,Rohan Mehta,Chandigarh,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.2,Debit Card
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card


Top 10 Orders by Quantity:


,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
26,1027,2026-05-02,Sara Ahmed,Chandigarh,North,Sports,Football,5,799,15,3395.75,4.6,Net Banking
24,1025,2026-01-02,Rohan Mehta,Chennai,South,Sports,Football,5,799,0,3995.00,5.0,UPI
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking
37,1038,2026-03-08,Aditya Verma,Pune,West,Sports,Dumbbell Set,5,1999,5,9495.25,5.0,Debit Card
71,1072,2026-05-03,Aarav Sharma,Jaipur,North,Sports,Yoga Mat,5,899,10,4045.50,4.3,Debit Card
72,1073,2026-06-04,Sara Ahmed,Kolkata,East,Books,Atomic Habits,5,599,15,2545.75,4.6,Credit Card
80,1081,2026-01-30,Reyansh Jain,Kolkata,East,Home & Kitchen,Electric Kettle,5,1499,15,6370.75,4.0,Debit Card
83,1084,2026-06-19,Aisha Mir,Kolkata,East,Electronics,Wireless Mouse,5,899,10,4045.50,3.8,Cash on Delivery
95,1096,2026-01-26,Aditya Verma,Kochi,South,Sports,Dumbbell Set,5,1999,5,9495.25,4.2,Credit Card
99,1100,2026-02-08,Kabir Ali,Pune,West,Books,Data Science Handbook,5,899,0,4495.00,4.8,Debit Card


In [5]:
# ============================================================
# 4. OVERALL SALES ANALYSIS
# ============================================================

total_sales = df["Total_Sales"].sum()
average_sales = df["Total_Sales"].mean()
highest_sales = df["Total_Sales"].max()
lowest_sales = df["Total_Sales"].min()
total_quantity = df["Quantity"].sum()
number_of_orders = df["Order_ID"].nunique()

print(f"Total Sales: ₹{total_sales:,.2f}")
print(f"Average Sales per Order: ₹{average_sales:,.2f}")
print(f"Highest Order Value: ₹{highest_sales:,.2f}")
print(f"Lowest Order Value: ₹{lowest_sales:,.2f}")
print(f"Total Quantity Sold: {total_quantity}")
print(f"Number of Orders: {number_of_orders}")

# Highest-value order
print("\nHighest Value Order:")
display(df.loc[df["Total_Sales"].idxmax()])

# Lowest-value order
print("\nLowest Value Order:")
display(df.loc[df["Total_Sales"].idxmin()])

# Orders above average sales
above_average = df[
    df["Total_Sales"] > df["Total_Sales"].mean()
]

print("\nOrders Above Average Sales:")
display(
    above_average[
        ["Order_ID", "Product", "City", "Total_Sales"]
    ].sort_values(
        by="Total_Sales",
        ascending=False
    ).head(10)
)

Total Sales: ₹447,262.95
Average Sales per Order: ₹4,472.63
Highest Order Value: ₹13,995.00
Lowest Order Value: ₹449.10
Total Quantity Sold: 294
Number of Orders: 100

Highest Value Order:


Order_ID                     1049
Order_Date             2026-03-02
Customer             Ishita Gupta
City                       Mumbai
Region                       West
Category                   Sports
Product             Running Shoes
Quantity                        5
Unit_Price                   2799
Discount_Percent                0
Total_Sales               13995.0
Rating                        4.0
Payment_Method        Net Banking
Name: 48, dtype: object


Lowest Value Order:


Order_ID                        1006
Order_Date                2026-01-21
Customer                   Manya Rao
City                           Kochi
Region                         South
Category                       Books
Product                The Alchemist
Quantity                           1
Unit_Price                       499
Discount_Percent                  10
Total_Sales                    449.1
Rating                           4.2
Payment_Method      Cash on Delivery
Name: 5, dtype: object


Orders Above Average Sales:


,Order_ID,Product,City,Total_Sales
48,1049,Running Shoes,Mumbai,13995.00
52,1053,Running Shoes,Hyderabad,13295.25
84,1085,Coffee Maker,Hyderabad,12596.40
89,1090,Cricket Bat,Lucknow,12495.00
12,1013,Jacket,Lucknow,10620.75
47,1048,Coffee Maker,Lucknow,10497.00
69,1070,Jacket,Delhi,9996.00
79,1080,Jacket,Delhi,9996.00
3,1004,Mechanical Keyboard,Lucknow,9496.20
51,1052,Mechanical Keyboard,Chandigarh,9496.20


In [6]:
# ============================================================
# 5. GROUPBY & AGGREGATION ANALYSIS
# ============================================================

# ---------------- CATEGORY-WISE ANALYSIS ----------------

category_analysis = df.groupby("Category").agg(
    Total_Sales=("Total_Sales", "sum"),
    Average_Sales=("Total_Sales", "mean"),
    Highest_Sale=("Total_Sales", "max"),
    Lowest_Sale=("Total_Sales", "min"),
    Total_Quantity=("Quantity", "sum"),
    Number_of_Orders=("Order_ID", "count")
).sort_values(
    by="Total_Sales",
    ascending=False
)

print("Category-wise Analysis:")
display(category_analysis)


# ---------------- CITY-WISE ANALYSIS ----------------

city_analysis = df.groupby("City").agg(
    Total_Sales=("Total_Sales", "sum"),
    Average_Sales=("Total_Sales", "mean"),
    Total_Quantity=("Quantity", "sum"),
    Number_of_Orders=("Order_ID", "count")
).sort_values(
    by="Total_Sales",
    ascending=False
)

print("City-wise Analysis:")
display(city_analysis)


# ---------------- PRODUCT-WISE ANALYSIS ----------------

product_analysis = df.groupby("Product").agg(
    Total_Sales=("Total_Sales", "sum"),
    Average_Sales=("Total_Sales", "mean"),
    Total_Quantity=("Quantity", "sum"),
    Number_of_Orders=("Order_ID", "count")
).sort_values(
    by="Total_Sales",
    ascending=False
)

print("Top 10 Products by Sales:")
display(product_analysis.head(10))


# ---------------- PAYMENT METHOD ANALYSIS ----------------

payment_analysis = df.groupby("Payment_Method").agg(
    Total_Sales=("Total_Sales", "sum"),
    Average_Sales=("Total_Sales", "mean"),
    Total_Quantity=("Quantity", "sum"),
    Number_of_Orders=("Order_ID", "count")
).sort_values(
    by="Total_Sales",
    ascending=False
)

print("Payment Method Analysis:")
display(payment_analysis)


# ---------------- REGION-WISE ANALYSIS ----------------

region_analysis = df.groupby("Region").agg(
    Total_Sales=("Total_Sales", "sum"),
    Average_Sales=("Total_Sales", "mean"),
    Total_Quantity=("Quantity", "sum"),
    Number_of_Orders=("Order_ID", "count")
).sort_values(
    by="Total_Sales",
    ascending=False
)

print("Region-wise Analysis:")
display(region_analysis)

Category-wise Analysis:


,Total_Sales,Average_Sales,Highest_Sale,Lowest_Sale,Total_Quantity,Number_of_Orders
Category,,,,,,
Sports,136026.45,5441.058000,13995.00,759.05,79,25
Home & Kitchen,113329.90,5396.661905,12596.40,1188.30,59,21
Electronics,92086.00,4846.631579,9496.20,1708.10,57,19
Clothing,69028.20,4930.585714,10620.75,679.15,39,14
Books,36792.40,1752.019048,4495.00,449.10,60,21


City-wise Analysis:


,Total_Sales,Average_Sales,Total_Quantity,Number_of_Orders
City,,,,
Hyderabad,59076.45,8439.492857,25,7
Chennai,47493.20,4749.320000,29,10
Lucknow,46194.65,7699.108333,22,6
Pune,41908.55,4190.855000,34,10
Mumbai,41457.80,5182.225000,23,8
Kochi,37257.50,4139.722222,29,9
Chandigarh,34539.65,4317.456250,22,8
Delhi,30882.80,5147.133333,18,6
Jaipur,25534.00,3647.714286,17,7


Top 10 Products by Sales:


,Total_Sales,Average_Sales,Total_Quantity,Number_of_Orders
Product,,,,
Coffee Maker,51610.25,7372.892857,16,7
Jacket,47106.15,9421.230000,20,5
Cricket Bat,41483.40,8296.680000,18,5
Running Shoes,41285.25,8257.050000,15,5
Wireless Headphones,33877.40,4839.628571,23,7
Dumbbell Set,30184.90,6036.980000,16,5
Mechanical Keyboard,26489.40,8829.800000,11,3
Electric Kettle,26382.40,5276.480000,19,5
Mixer Grinder,25341.55,6335.387500,9,4


Payment Method Analysis:


,Total_Sales,Average_Sales,Total_Quantity,Number_of_Orders
Payment_Method,,,,
Credit Card,106540.25,4842.738636,65,22
Debit Card,98064.90,4903.245000,64,20
Net Banking,96371.25,5668.897059,63,17
UPI,96163.25,3698.586538,70,26
Cash on Delivery,50123.30,3341.553333,32,15


Region-wise Analysis:


,Total_Sales,Average_Sales,Total_Quantity,Number_of_Orders
Region,,,,
North,164989.45,4341.827632,102,38
South,157351.45,5245.048333,97,30
West,98155.05,4089.793750,69,24
East,26767.00,3345.875000,26,8


In [7]:
# ============================================================
# 6. DATE & TOP-PERFORMER ANALYSIS
# ============================================================

# Convert Order_Date to datetime
df["Order_Date"] = pd.to_datetime(df["Order_Date"])

# Extract month
df["Month"] = df["Order_Date"].dt.month

# Monthly sales
monthly_analysis = df.groupby("Month").agg(
    Total_Sales=("Total_Sales", "sum"),
    Number_of_Orders=("Order_ID", "count"),
    Total_Quantity=("Quantity", "sum")
).sort_values(
    by="Total_Sales",
    ascending=False
)

print("Monthly Sales Analysis:")
display(monthly_analysis)


# Top-performing category
top_category = (
    df.groupby("Category")["Total_Sales"]
    .sum()
    .idxmax()
)

# Top-performing product
top_product = (
    df.groupby("Product")["Total_Sales"]
    .sum()
    .idxmax()
)

# Top-performing city
top_city = (
    df.groupby("City")["Total_Sales"]
    .sum()
    .idxmax()
)

# Top payment method
top_payment = (
    df.groupby("Payment_Method")["Total_Sales"]
    .sum()
    .idxmax()
)

print("Top Performing Category:", top_category)
print("Top Performing Product:", top_product)
print("Top Performing City:", top_city)
print("Top Payment Method:", top_payment)

Monthly Sales Analysis:


,Total_Sales,Number_of_Orders,Total_Quantity
Month,,,
3,126311.15,22,68
1,79539.40,18,55
5,70267.00,17,49
6,69924.75,17,54
4,59386.95,15,35
2,41833.70,11,33


Top Performing Category: Sports
Top Performing Product: Coffee Maker
Top Performing City: Hyderabad
Top Payment Method: Credit Card


# KEY OBSERVATIONS & CONCLUSION

## Key Observations

- The dataset contains 100 e-commerce orders.
- Total sales generated are ₹447,262.95.
- The average sales value per order is approximately ₹4,472.63.
- A total of 294 items were sold.
- Sports is the highest-performing category based on total sales.
- Coffee Maker is the top-performing product based on total sales.
- Hyderabad generated the highest city-wise sales.
- Credit Card generated the highest sales among the payment methods.
- High-value orders can be identified using filtering and sorting operations.
- GroupBy and aggregation provide useful insights into category, city,
  product, region, and payment-method performance.

## Conclusion

The analysis demonstrates how Pandas can be used to explore and analyze
e-commerce sales data efficiently. Data selection, filtering, sorting,
groupby(), and aggregation functions help identify important sales trends,
top-performing products and categories, high-value orders, and customer
purchasing patterns.